Commonalities/documentation/API-design-guidelines.md

```diff
@@ -1021,3 +1027,180 @@ Finally, this part describes the OAuth security applied to the API. This spec is
 - Endpoint token URL
 - URL to endpoint authorization ( If flow is based on "`authorizationCode`").	
 
+
+## 12. Subscription, Notification & Event
+
+In order to provided event-based interaction, CAMARA API could provided capabilities for subscription & notification management.
+A subscription allows to an API consumer to request event notification reception at an given url (callback address) for a specific context.
+A notification is the publication at the listener address about an occurred event.
+Event type managed are explicitly defined in CAMAPA API OAS.
+
+### 12.1 Subscription
+
+We distinguish 2 types of subscription:
+- Instance-based subscription (indirect creation)
+- Resource-based subscription (direct creation)
+
+**Instance-based subscription**
+
+An instance-based subscription is a subscription indirectly created, additionnally, to another resource creation. For example for a Payment request (in Carrier Billing API), in the `POST/payments`, the API consumer could request to get event notification about **this** Payment request processing update. The subscription is not an entity and its lifecycle is linked to the managed entity (the Payment resource in this case). The subscription terminates with the managed entity.
```

Comment:
The use of this option, ¿overwrites an existing subscription, generated by means of resource-based creation? Think we need to ensure how to address scenarios where both options may be available for a given API

Reply:
For me, if an API provided both pattern, and an API customer request both (instance base + subscription) we should handle them independently & autonomously. Providing rules overwriting one above the other can add complexity without real value added no ?

Reply:
Need to think about this, because we can add high complexity to the logic of the notifications server

Reply:
Yes sure !
For me managing them separately both in server side as customer side as separate transaction is the easiest pattern but please take time to think about it.